testsuite/features/step_definitions/command_steps.rb

```diff
@@ -1563,3 +1563,30 @@
   raise 'File injection failed' unless return_code.zero?
   node.run('salt-call --local --file-root=/usr/share/susemanager/salt --module-dirs=/usr/share/susemanager/salt/ --log-level=info --retcode-passthrough state.apply ' + state)
 end
+
+When(/^I copy autoinstall mocked files on server$/) do
+  target_dirs = %w[/autoinstall/Fedora_12_i386/images/pxeboot/ /autoinstall/SLES15-SP2-x86_64/DVD1/boot/x86_64/loader/ /autoinstall/mock]
```

Comment:
Simplify and accelerate: no loop, no when :-) .

1) only one `mkdir` at the beginning
2) as many `file_insert()` as needed in the middle
3) if really needed, only one `chmod` at the end

as for error detection, you could for example gather all error codes in one array. I think it's a one-liner in ruby to see if any of them is nonzero.

Reply:
So, your method implementation is perfectly fine IMHO an is readable and maintainable.
The worry of Eric, and this is important worry for us, is that we have 1200 scenarios in our test suite and we try to speed up our tests as much as we can. Every call to `$server.run` is a connection to the server, so if we can wrap some actions in a unique call we speed up the process. And that's why Eric insist that much ;)

Reply:
With the caasp project I was keeping the ssh sections open to save some time during ssh command execution.
It added some complicity to keep it alive correctly and refresh it after reboot.
I think it's probably too complicated to add for this project but could be interesting for the new automation project.